### DISCUSSION TITLE
[HELP] Radio communication

### DISCUSSION THREAD

**Discussion by fp018** [bitcraze/discussions#1408] [Q&A] [answered]
**ISSUE**
I am having trouble flying more crazyflies at the same time (even 3) and I am wondering if it is an issue related to radio communication. I have noticed that in the swam example folder radios are configured in two different ways.

**1. One radio and one channel:**

      URI1 = 'radio://0/70/2M/E7E7E7E701'
      URI2 = 'radio://0/70/2M/E7E7E7E702'
      URI3 = 'radio://0/70/2M/E7E7E7E703' 
      [..]

**2. One radio and multiple channels:**

      URI0 = 'radio://0/70/2M/E7E7E7E7E7'
      URI1 = 'radio://0/110/2M/E7E7E7E702'
      URI2 = 'radio://0/94/2M/E7E7E7E7E7'
      [..]

I have been using the **option 1. so far**, but I have just found out in the docs (https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/) that:

> If using 2M datarate, the copter channels should be 2 apart (2MHz).

Can anyone clarify on this point? 

Thanks in advance!

**Comment by knmcguire** [comment]
I've just converted this to a discussion.

You are on the right path of using the same channel. I'm not sure why the userguide of the client says that, but I think that is just to indicate if you are setting up different channels for each of the different crazyflie it is best to stay 2 MHz apart for the channel. But as long as they have different addresses on the same channel it is fine.

I don't have issues with the cflib with 3 crazyflies at one time. Which script are you using? Maybe you are hitting a communication bottle neck.

**Reply by fp018** [reply]
Thank you very much for the reply. 
I am using a modified version of the python crazyflie server of **crazyswarm2** repo, and I am having trouble even with two drones on a single laptop, as soon as I add the second the light of the radio starts blinking. I am commanding full state setpoints and logging at the same time.

**Reply by knmcguire** [reply]
Alright. technically a question for Crazyswarm2 I'm afraid but this sounds like a communication bottle neck.

What positioning are you using and at what rate are you sending the full state setpoint and receiving logging variables

**Reply by fp018** [reply]
I am using Qualisys Motion capture with 100Hz/150Hz capture frequency, publishing body’s poses. I am also logging _/odom_ from crazy flies at 30Hz, but I had the same issue also disabling the logging.

**Reply by knmcguire** [reply, marked as answer]
So, one crazyradio has the capability of sending and receiving out 400ish packets per second, and those are under ideal situation. So if you are sending external position wiht 150 hz, and want to do logging as well, you are hitting the limit for sure. You'll need to tune down the frequency of sending the external position way down because this is hitting the communication bottle neck for sure.
